tokio/Cargo.toml

```diff
@@ -132,7 +132,7 @@ nix = { version = "0.26", default-features = false, features = ["fs", "socket"]
 version = "0.48"
 optional = true
 
-[target.'cfg(docsrs)'.dependencies.windows-sys]
+[target.'cfg(windows)'.dev-dependencies.windows-sys]
 version = "0.48"
 features = [
     "Win32_Foundation",
```

Comment:
What happened here?

Reply:
In the v1 resolver, the features were merged in all `target.'...'.dependencies`, but in the v2 resolver, only the currently valid ones are considered. As a result, I got errors that `windows_sys::Win32::Security::Authorization` does not exist with `cargo test` that did not set `docsrs` cfg.

https://github.com/tokio-rs/tokio/actions/runs/5233722144/jobs/9449397281

```
---- src\net\windows\named_pipe.rs - net::windows::named_pipe::ServerOptions::write_dac (line 2064) stdout ----
error[E0432]: unresolved import `windows_sys::Win32::Security::Authorization`
 --> src\net\windows\named_pipe.rs:2070:22
  |
8 |     Win32::Security::Authorization::{SetSecurityInfo, SE_KERNEL_OBJECT},
  |                      ^^^^^^^^^^^^^ could not find `Authorization` in `Security`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
Couldn't compile the test.
---- src\net\windows\named_pipe.rs - net::windows::named_pipe::ServerOptions::write_dac (line 2098) stdout ----
error[E0432]: unresolved import `windows_sys::Win32::Security::Authorization`
 --> src\net\windows\named_pipe.rs:2104:22
  |
8 |     Win32::Security::Authorization::{SetSecurityInfo, SE_KERNEL_OBJECT},
  |                      ^^^^^^^^^^^^^ could not find `Authorization` in `Security`
```

Reply:
Should the feature not also be enabled for docsrs? Or is it just a mistake that its listed under docsrs?

Reply:
We currently use `windows_sys::Win32::Security::Authorization` for only two doctests so `'cfg(docsrs)'.dependencies.windows-sys` should not be necessary.

https://github.com/tokio-rs/tokio/blob/2a54ad01d0945c851a093849c83019de69e98dee/tokio/src/net/windows/named_pipe.rs#L2071
https://github.com/tokio-rs/tokio/blob/2a54ad01d0945c851a093849c83019de69e98dee/tokio/src/net/windows/named_pipe.rs#L2106

As for `windows_sys::Win32::Foundation`, it seems that all features that depend on windows-sys enable it.

https://github.com/tokio-rs/tokio/blob/2a54ad01d0945c851a093849c83019de69e98dee/tokio/Cargo.toml#L55
https://github.com/tokio-rs/tokio/blob/2a54ad01d0945c851a093849c83019de69e98dee/tokio/Cargo.toml#L68
https://github.com/tokio-rs/tokio/blob/2a54ad01d0945c851a093849c83019de69e98dee/tokio/Cargo.toml#L84